SgmlTests/Tests-Logic.cs

```diff
@@ -9,143 +9,168 @@
+                {
+                    return sr.ReadToEnd();
+                }
             }
         }
 
-        private static string RunTest(CaseFolding caseFolding, string doctype, bool format, string source, XmlReaderTestCallback callback) {
+        /// <summary></summary>
+        /// <param name="docType"></param>
+        /// <param name="name">The name of the EmbeddedResource file located under the <c>SgmlTests\Resources</c> project directory.</param>
+        /// <returns></returns>
+        internal static SgmlDtd LoadDtd(string docType, string resourceFileName)
+        {
+            StringReader dtdTextReader = new StringReader(ReadTestResource(resourceFileName));
+
+            SgmlDtd dtd = SgmlDtd.Parse(baseUri: null, name: docType, input: dtdTextReader, subset: "", nt: new NameTable(), resolver: new DesktopEntityResolver());
+            return dtd;
+        }
+
+        private static string RunTest(CaseFolding caseFolding, string doctype, bool format, string source, XmlReaderTestCallback callback)
+        {
 
             // initialize sgml reader
-            XmlReader reader = new SgmlReader {
+            SgmlReader sgmlReader = new SgmlReader
+            {
                 CaseFolding = caseFolding,
                 DocType = doctype,
                 InputStream = new StringReader(source),
                 WhitespaceHandling = format ? WhitespaceHandling.None : WhitespaceHandling.All
             };
+
             if (doctype == "OFX")
             {
-                ((SgmlReader)reader).Dtd = LoadDtd("OFX", "ofx160.dtd");
+                sgmlReader.Dtd = LoadDtd("OFX", "ofx160.dtd");
             }
 
-
             // check if we need to use the LoggingXmlReader
-            if (_debug) {
-                reader = new LoggingXmlReader(reader, Console.Out);
+            XmlReader reader = sgmlReader;
+#if DEBUG
+            {
+                reader = new LoggingXmlReader(sgmlReader, Console.Out);
             }
+#endif
 
             // initialize xml writer
-            var stringWriter = new StringWriter();
-            var xmlTextWriter = new XmlTextWriter(stringWriter);
-            if(format) {
+            StringWriter stringWriter = new StringWriter();
+            XmlTextWriter xmlTextWriter = new XmlTextWriter(stringWriter);
+            if (format)
+            {
                 xmlTextWriter.Formatting = Formatting.Indented;
             }
             callback(reader, xmlTextWriter);
             xmlTextWriter.Close();
 
             // reproduce the parsed document
-            var actual = stringWriter.ToString();
+            string actual = stringWriter.ToString();
 
             // ensure that output can be parsed again
-            try {
-                using(var stringReader = new StringReader(actual)) {
-                    var doc = new XmlDocument();
+            try
+            {
+                using (StringReader stringReader = new StringReader(actual))
```

Comment:
why are you removing "var".  "var" is a good thing.

Reply:
This was also done by auto-formatting (I'm a huge _opponent_ to `var`, FWIW ;) - I thought the `.editorconfig` file was configured to use `var` though, I'll update the rules and undo these changes in particular. Sorry for the noise!

Reply:
Thanks.  I'm a huge proponent of var because unlike javascript, var in C# is still strongly typed.  You lose no strong typing, and you gain some code robustness in the face of some kinds of renaming changes.  Now I do agree in some cases it is nice to see the typename there if it is not obvious from the right hand side, so that's why the code doesn't exclusively use var.  So I'm sort of in the middle, "use var when it is obvious while reading the code what the type is".